core/src/Kokkos_CopyViews.hpp

```diff
@@ -1506,12 +1507,743 @@ void deep_copy
+    #pragma unroll
+    for(int i=0;i<N;++i){
+        int i0   = i%dst.extent(0);
+        int itmp = i/dst.extent(0);
+        int i1   = itmp%dst.extent(1);
+            itmp = itmp/dst.extent(1);
+        int i2   = itmp%dst.extent(2);
+            itmp = itmp/dst.extent(2);
+        int i3   = itmp%dst.extent(3);
+            itmp = itmp/dst.extent(3);
+        int i4   = itmp%dst.extent(4);
+            itmp = itmp/dst.extent(4);
+        int i5   = itmp%dst.extent(5);
+        int i6   = itmp/dst.extent(5);
+        dst(i0,i1,i2,i3,i4,i5,i6) = src(i0,i1,i2,i3,i4,i5,i6);
+    }
+}
+//----------------------------------------------------------------------------
+template< class DT , class ... DP , class ST , class ... SP >
+void KOKKOS_INLINE_FUNCTION local_deep_copy (const View<DT,DP...> & dst,  
+                                             const View<ST,SP...> & src,
+                                             typename std::enable_if<( (unsigned(ViewTraits<DT,DP...>::rank) != 0 && 
+                                                                        unsigned(ViewTraits<ST,SP...>::rank) != 0) &&    
+                                                                        std::is_same< typename View<DT,DP...>::memory_space, Kokkos::HostSpace >::value
+                                                                     )>::type * = 0 )
+{
+    if(dst.data() == NULL ) {
+        //Kokkos::fence();
+        return;
+    }
+    Kokkos::deep_copy(dst,src); //printf("local_deep_copy: rangepolicy, rank !=0, use Kokkos:deep_copy on host memspace\n");
+}
+//----------------------------------------------------------------------------
+/** \brief  Deep copy a value into a view.  */
+template< class TeamType, class DT , class ... DP >
+void KOKKOS_INLINE_FUNCTION local_deep_copy (const TeamType& team, const View<DT,DP...> & dst, 
+                                                                   typename ViewTraits<DT,DP...>::const_value_type & value,
+                                                                   typename std::enable_if<( unsigned(ViewTraits<DT,DP...>::rank) == 1 )>::type * = 0 )
+{
+     if(dst.data() == NULL ) {
+        //Kokkos::fence();
+        return;
+    }
+    const int N = dst.extent(0); //if(team.team_rank() == 0) //printf("local_fill: teampolicy, rank 1\n");
+    team.team_barrier();
+    Kokkos::parallel_for(Kokkos::TeamThreadRange(team, N), [&] (const int& i) {
+        dst(i) =  value;
+    });
+    team.team_barrier();
+}
+//----------------------------------------------------------------------------
+template< class TeamType, class DT , class ... DP >
+void KOKKOS_INLINE_FUNCTION local_deep_copy (const TeamType& team, const View<DT,DP...> & dst, 
+                                                                   typename ViewTraits<DT,DP...>::const_value_type & value,
```

Comment:
@mhoemmen To avoid ```warning: calling a host function from a host device function```, I will not use host functions such as ```memset``` or ```memcpy```, instead, I added specializations for contiguous Views using ```for``` and ```parallel_for```.